### PR TITLE
Cheap eqs: remove the table approach

### PR DESCRIPTION
**Pull request by levnach**

### CODE REVIEW

File: src/smt/theory_lra.cpp [hunk 1]
```diff
@@ -1395,10 +1395,22 @@ class theory_lra::imp {
 #endif
         }
         else {
-            literal div_ge_0   = mk_literal(a.mk_ge(div, zero));
-            literal div_le_0   = mk_literal(a.mk_le(div, zero));
-            literal p_ge_0     = mk_literal(a.mk_ge(p, zero));
-            literal p_le_0     = mk_literal(a.mk_le(p, zero));
+#if 0
```
Comment (NikolajBjorner) [inline src/smt/theory_lra.cpp:1398]: seems these literals shouldn't be created?

Comment (levnach) [reply src/smt/theory_lra.cpp:1398]: I am not sure. These literals used in code below which is surrounded by #if 0 ... #endif

File: src/math/lp/hnf_cutter.h [hunk 2]
```diff
@@ -8,7 +8,8 @@ Module Name:
 Abstract:
 
     Cuts (branches) from Hermite matrices
-
+    The implementation is based on ideas from
+    "Cutting the Mix" by Jurgen Christ and Jochen Hoenicke.
```
Comment (NikolajBjorner) [inline src/math/lp/hnf_cutter.h:12]: and Cuts from Proofs: A Complete and Practical Technique for Solving Linear Inequalities over Integers
Isil Dillig, Thomas Dillig, Alex Aiken. Formal Methods in System Design (FMSD) CAV 2009 special issue.

File: src/math/lp/lp_bound_propagator.h [hunk 3]
```diff
@@ -278,35 +233,74 @@ class lp_bound_propagator {
         }
         return false;
     }
+
+    // pol for polarity
+    int pol(const vertex* v) const { return pol(v->column()); }
+    int pol(unsigned j) const { return m_pol[j].pol(); }
```
Comment (NikolajBjorner) [inline src/math/lp/lp_bound_propagator.h:239]: is polarity a sign as in util/sign.h?

Comment (levnach) [reply src/math/lp/lp_bound_propagator.h:239]: Yes, it is almost the same, but cannot be zero. I am not sure using sign.h will simplify things.
For example, the code 
` if (pol(v->column()) == -1)`
would be replaced by 
`if (pol(v->column() == sign_neg)`.

**Pull request merged**